oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandler.java

```diff
@@ -201,22 +241,123 @@ public APIGatewayProxyResponseEvent authoriseRequestHandler(
                     persistentSessionId,
                     authRequest.getClientID().getValue(),
                     clientSessionId);
-        } else {
-            authRequest = RequestObjectToAuthRequestHelper.transform(authRequest);
-            return getOrCreateSessionAndRedirect(
-                    sessionService.getSessionFromSessionCookie(input.getHeaders()),
+        }
+        Optional<Session> session = sessionService.getSessionFromSessionCookie(input.getHeaders());
+        if (configurationService.isDocAppDecoupleEnabled()
+                && DocAppUserHelper.isDocCheckingAppUser(
+                        authRequest.toParameters(), Optional.of(client))) {
+            return handleDocAppJourney(
+                    session,
                     authRequest,
+                    client,
+                    clientSessionId,
                     ipAddress,
-                    persistentSessionId,
-                    clientSessionId);
+                    PersistentIdHelper.extractPersistentIdFromHeaders(input.getHeaders()));
         }
+
+        authRequest = RequestObjectToAuthRequestHelper.transform(authRequest);
+        return handleAuthJourney(
+                session, authRequest, ipAddress, persistentSessionId, client, clientSessionId);
+    }
+
+    private APIGatewayProxyResponseEvent handleDocAppJourney(
+            Optional<Session> existingSession,
+            AuthenticationRequest authRequest,
+            ClientRegistry client,
+            String clientSessionId,
+            String ipAddress,
+            String persistentId) {
+
+        var session = existingSession.orElseGet(sessionService::createSession);
+        attachSessionIdToLogs(session);
+
+        if (existingSession.isEmpty()) {
+            updateAttachedSessionIdToLogs(session.getSessionId());
+            LOG.info("Created session");
+        } else {
+            var oldSessionId = session.getSessionId();
+            sessionService.updateSessionId(session);
+            updateAttachedSessionIdToLogs(session.getSessionId());
+            LOG.info("Updated session id from {} - new", oldSessionId);
+        }
```

Comment:
This looks to be duplicated in here and the `handleAuthJourney` method. Wonder if it's worth putting this before either the `handleDocAppJourney` or the `handleAuthJourney` methods are invoked to avoid this duplication